SilKit/ci/check_formatting.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2024 Vector Informatik GmbH
+#
+# SPDX-License-Identifier: MIT
+from pathlib import Path
+from shutil import which
+import re
+import subprocess
+import sys
+
+## Convenience
+def log(fmt, *args):
+    print(fmt.format(*args))
+
+def info(fmt, *args):
+    log("INFO: " + fmt, *args)
+
+def warn(fmt, *args):
+    log("::warning:: " + fmt, *args)
+
+def die(status, fmt, *args):
+    log("ERROR: " + fmt, *args)
+    sys.exit(status)
+
+## Check if clang-format is installed
+
+if which('clang-format') is None:
+    warn("No clang-format found!")
+    die(1, "Please install clang-format!")
+
+# Check for supported clang-format version
+format_version = subprocess.run(['clang-format', '--version'], capture_output=True)
+
+version_reg = re.compile('^.* clang-format version (\d+)\.(\d+)\.(\d+).*')
+version = re.match(version_reg, format_version.stdout.decode('utf-8'))
+
+if version is None or len(version.groups()) != 3:
+    die(2, "ERROR: Could not get the clang-format version!")
+
+major, minor, patch = version.group(1,2,3)
+if int(major) < 13:
+    die(3, "clang{} not supported!\r\n       Minimum supported version is clang14!", major)
+
+info("clang-format-{}.{}.{} found!", major, minor, patch)
+
+############################################################################################################################
+fileExtensions = ["*.cpp", "*.ipp", "*.c", "*.hpp", "*.h"]
+dirs = ["SilKit", "Demos", "Utilities"]
+
+formatting_correct = True
+
+# Open LogFile to get some data on our poorly formatted files!
+with open('FormatLog.txt', 'w') as fLog:
+
+# Check  the Formatting!
+    for directory in dirs:
+        rootPath = Path("./" + directory)
+        for ext in fileExtensions:
+            files = sorted(rootPath.rglob(ext))
+            #print("INFO: Found {} {} files in {}!".format(len(files), ext, directory))
+
+            for file in files:
+                format_result = subprocess.run(['clang-format', '--Werror', '--dry-run', '-i', '--style=file', file], capture_output=True)
+                if format_result.returncode != 0:
+                    formatting_correct = False
+                    fLog.write("{}\n".format(file));
```

Comment:
To me it makes sense to not only log the files, but also print them to stdout or stderr. This may be a long list (for huge commits / PRs), but since changes should be relatively small (a few files) you won't have to download a file, but can just read it off GitHub.

Reply:
Also: you can write to files in python using `print("some string", file=fLog)` which gives you the print-style formatting, but into any file stream.

Just thought I'd mention it - not really better or worse imho.

Reply:
Yeah, @MariusBgm mentioned something similar. I will definitely change to a `just log to console` presentation of the files. I guess I didn't want to spam the console as long as  https://github.com/vectorgrp/sil-kit/pull/37 is not merged yet :D